storage-proofs/benches/preprocessing.rs

```diff
@@ -51,16 +52,15 @@ fn preprocessing_benchmark(c: &mut Criterion) {
     );
 }
 
-fn write_padded_bench(file: &mut File, data: &[u8]) {
-    write_padded(&data, file).unwrap();
-
+fn write_padded_bench(file: &mut File, mut data: Vec<u8>) {
```

Comment:
Might be nicer to pass in `&mut [u8]` instead of giving it the whole vec, that way you can avoid the clones

Reply:
I need the clone because the closure in which `write_padded_bench` is called is `FnMut` and not `FnOnce`. I'd have to use an `Rc` (or some other reference counted type) to avoid the clone. I'm not particularly concerned about memory efficiency in the benchmark setup.

Reply:
well the issue is that it dilutes the benchmark, because now you are benchmarking how long it takes to clones those bytes

Reply:
Why does that matter? This benchmark is for capturing deltas in performance of the bit padding operation across (code) changes to the operation itself.

Reply:
Putting that aside for a moment: 

Note that `Read#read` is of type `read(&mut self, buf: &mut [u8]) -> Result<usize>`. Given that type, I see no way around cloning the vector.